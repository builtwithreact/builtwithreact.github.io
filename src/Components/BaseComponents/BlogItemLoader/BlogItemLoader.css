.BlogItemLoader {
  display: flex;
  height: 204px; }
  @media screen and (max-width: 767px) {
    .BlogItemLoader {
      height: 100px; } }
  .BlogItemLoader .BlogItemLoader__Image {
    background: #c3c2c2;
    animation-name: loading;
    animation-duration: 1s;
    animation-iteration-count: infinite;
    width: 250px;
    height: 100%;
    margin-right: 40px; }

@keyframes loading {
  0% {
    opacity: 0.9; }
  25% {
    opacity: 0.7; }
  50% {
    opacity: 0.5; }
  75% {
    opacity: 0.7; }
  100% {
    opacity: 0.9; } }
    @media screen and (max-width: 767px) {
      .BlogItemLoader .BlogItemLoader__Image {
        width: 100px;
        margin-right: 20px; } }
  .BlogItemLoader .BlogItemLoader__Body {
    display: flex;
    flex-direction: column;
    width: 100%;
    flex: 1; }
  .BlogItemLoader .BlogItemLoader__Body__Title {
    background: #c3c2c2;
    animation-name: loading;
    animation-duration: 1s;
    animation-iteration-count: infinite;
    height: 40px; }

@keyframes loading {
  0% {
    opacity: 0.9; }
  25% {
    opacity: 0.7; }
  50% {
    opacity: 0.5; }
  75% {
    opacity: 0.7; }
  100% {
    opacity: 0.9; } }
  .BlogItemLoader .BlogItemLoader__Body__Content {
    background: #c3c2c2;
    animation-name: loading;
    animation-duration: 1s;
    animation-iteration-count: infinite;
    margin-top: 20px;
    height: 70px; }

@keyframes loading {
  0% {
    opacity: 0.9; }
  25% {
    opacity: 0.7; }
  50% {
    opacity: 0.5; }
  75% {
    opacity: 0.7; }
  100% {
    opacity: 0.9; } }
  .BlogItemLoader .BlogItemLoader__Body__Content__Link {
    margin-top: auto;
    text-transform: uppercase; }
    .BlogItemLoader .BlogItemLoader__Body__Content__Link .BlogItemLoader__Body__Content__Link__a {
      background: #c3c2c2;
      animation-name: loading;
      animation-duration: 1s;
      animation-iteration-count: infinite;
      width: 100px;
      height: 15px;
      margin: 15px 0 5px; }

@keyframes loading {
  0% {
    opacity: 0.9; }
  25% {
    opacity: 0.7; }
  50% {
    opacity: 0.5; }
  75% {
    opacity: 0.7; }
  100% {
    opacity: 0.9; } }
