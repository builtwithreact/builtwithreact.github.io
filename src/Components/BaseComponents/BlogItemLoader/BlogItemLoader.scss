@import './../../../Assets/Styles/skeletion';

.BlogItemLoader {
  display: flex;
  height: 204px;
  @media screen and (max-width: 767px) { 
    height: 100px;
  }

  .BlogItemLoader__Image {
    @include skeletion();
    width: 250px;
    height: 100%;
    margin-right: 40px;
    @media screen and (max-width: 767px) { 
      width: 100px;
      margin-right: 20px;
    }
  }

  .BlogItemLoader__Body{
    display: flex;
    flex-direction: column;
    width: 100%;
    flex: 1;
  }

  .BlogItemLoader__Body__Title {
    @include skeletion();
    height: 40px;
  }

  .BlogItemLoader__Body__Content {
    @include skeletion();
    margin-top: 20px;
    height: 70px;
  }

  .BlogItemLoader__Body__Content__Link {
      // border-top: 1px #ccc solid;
      margin-top: auto;
      text-transform: uppercase;
      .BlogItemLoader__Body__Content__Link__a {
        @include skeletion();
        width: 100px;
        height: 15px;
        margin: 15px 0 5px;
      }
  }
}