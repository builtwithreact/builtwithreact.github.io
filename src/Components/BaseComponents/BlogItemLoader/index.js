import React from 'react';
import './BlogItemLoader.css';

const BlogItemLoader = () => {
    return (
        <div className="BlogItemLoader">
            <div className="BlogItemLoader__Image" />
            <div className="BlogItemLoader__Body">
                <div className="BlogItemLoader__Body__Title" />
                <div className="BlogItemLoader__Body__Content" />
                <div className="BlogItemLoader__Body__Content__Link">
                    <div className="BlogItemLoader__Body__Content__Link__a" />
                </div>
            </div>
        </div>
    )
}

export default BlogItemLoader;