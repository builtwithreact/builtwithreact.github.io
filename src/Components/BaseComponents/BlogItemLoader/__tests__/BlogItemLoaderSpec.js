import React from 'react';
import renderer from 'react-test-renderer';
import BlogItemLoader from './../index';

describe("<BlogItemLoader />", () => {
    it('should render correctly', ()=>{
        let tree = renderer.create(
            <BlogItemLoader />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })
})