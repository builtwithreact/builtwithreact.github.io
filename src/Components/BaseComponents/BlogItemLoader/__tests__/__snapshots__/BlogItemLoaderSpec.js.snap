// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogItemLoader /> should render correctly 1`] = `
<div
  className="BlogItemLoader"
>
  <div
    className="BlogItemLoader__Image"
  />
  <div
    className="BlogItemLoader__Body"
  >
    <div
      className="BlogItemLoader__Body__Title"
    />
    <div
      className="BlogItemLoader__Body__Content"
    />
    <div
      className="BlogItemLoader__Body__Content__Link"
    >
      <div
        className="BlogItemLoader__Body__Content__Link__a"
      />
    </div>
  </div>
</div>
`;
